spacy/cli/project/assets.py

```diff
@@ -61,7 +61,7 @@ def project_assets(project_dir: Path, *, sparse_checkout: bool = False) -> None:
                         dest.unlink()
             git_checkout(
                 asset["git"]["repo"],
-                asset["git"]["path"],
+                asset["git"].get("path", ""),
```

Comment:
I don't think we should have these kind of default values defined deep within the code, it makes it difficult to track where defaults are getting introduced. Could we instead just check whether `subpath is None` in `git_checkout` and raise a custom error there? In the error message we could specify that `""` can be used to denote the entire directory.

Reply:
A couple of issues I see there -- this would violate the type signature of `git_checkout` by passing None in where a str is expected (unless our types are implicitly optional?), and also we'd have a less pertinent error message because it'd be referring to the `subpath` parameter of `git_checkout` instead of the "path" configuration option.

But I could check for this and raise a custom error in `project_assets` instead.